storage/innobase/xtrabackup/src/keyring_plugins.cc

```diff
@@ -328,28 +402,53 @@ xb_keyring_init_for_prepare(int argc, char **argv)
 		return(false);
 	}
 
-	if (plugin_load != NULL) {
-		opt_plugin_load_list_ptr->push_back(
-			new i_string(my_strdup(
-				PSI_NOT_INSTRUMENTED, plugin_load,
-				MYF(MY_FAE))));
-	}
-
 	memset(server_uuid, 0, ENCRYPTION_SERVER_UUID_LEN + 1);
 	if (uuid != NULL) {
 		strncpy(server_uuid, uuid, ENCRYPTION_SERVER_UUID_LEN);
 	}
 
-	int t_argc = argc;
+	/* copy argc, argv because handle_options will destroy them */
+
+	int t_argc = argc + 1;
 	char **t_argv = new char *[t_argc + 1];
 
 	memset(t_argv, 0, sizeof(char *) * (t_argc + 1));
-	memcpy(t_argv, argv, sizeof(char *) * t_argc);
+	t_argv[0] = exename;
+	memcpy(t_argv + 1, argv, sizeof(char *) * argc);
+
+	my_option plugin_load_options[] = {
+		{"early-plugin-load", 1, "", 0, 0, 0, GET_STR, REQUIRED_ARG,
+		 0, 0, 0, 0, 0, 0},
+		{"plugin-load", 2, "", 0, 0, 0, GET_STR, REQUIRED_ARG,
+		 0, 0, 0, 0, 0, 0},
+		{"plugin-load-add", 3, "", 0, 0, 0, GET_STR, REQUIRED_ARG,
+		 0, 0, 0, 0, 0, 0},
+		{0, 0, 0, 0, 0, 0, GET_NO_ARG, NO_ARG, 0, 0, 0, 0, 0, 0}
+	};
 
-	plugin_init(&t_argc, t_argv, PLUGIN_INIT_SKIP_PLUGIN_TABLE);
+	char **old_t_argv = t_argv;
+	if (handle_options(&t_argc, &t_argv, plugin_load_options,
+			   get_plugin_load_option)) {
+		return(false);
```

Comment:
Shouldn't we delete `old_t_argv` here to be consistent?

Reply:
I guess we should :)